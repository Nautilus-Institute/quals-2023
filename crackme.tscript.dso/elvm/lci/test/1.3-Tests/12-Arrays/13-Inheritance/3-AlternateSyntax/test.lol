HAI 1.3
	I HAS A parent ITZ A BUKKIT
	parent HAS A foo ITZ 1
	parent HAS A bar ITZ 2.34
	parent HAS A baz ITZ FAIL
	parent HAS A qux ITZ "HAI"

	VISIBLE parent 'Z foo
	VISIBLE parent 'Z bar
	VISIBLE SUM OF 0 AN parent 'Z baz
	VISIBLE parent 'Z qux

	O HAI IM child IM LIEK parent
		VISIBLE foo
		VISIBLE bar
		VISIBLE SUM OF 0 AN baz
		VISIBLE qux

		I HAS A foo ITZ 9
		I HAS A bar ITZ 8.76
		I HAS A baz ITZ WIN
		I HAS A qux ITZ "BAI"

		VISIBLE foo
		VISIBLE bar
		VISIBLE SUM OF 0 AN baz
		VISIBLE qux

		VISIBLE parent 'Z foo
		VISIBLE parent 'Z bar
		VISIBLE SUM OF 0 AN parent 'Z baz
		VISIBLE parent 'Z qux
	KTHX

	VISIBLE child 'Z foo
	VISIBLE child 'Z bar
	VISIBLE SUM OF 0 AN child 'Z baz
	VISIBLE child 'Z qux

	VISIBLE parent 'Z foo
	VISIBLE parent 'Z bar
	VISIBLE SUM OF 0 AN parent 'Z baz
	VISIBLE parent 'Z qux

KTHXBYE
