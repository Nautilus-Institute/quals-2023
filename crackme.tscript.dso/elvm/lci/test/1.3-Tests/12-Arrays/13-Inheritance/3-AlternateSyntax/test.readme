This test is designed to test inheritance for arrays declared using the
alternate syntax.
