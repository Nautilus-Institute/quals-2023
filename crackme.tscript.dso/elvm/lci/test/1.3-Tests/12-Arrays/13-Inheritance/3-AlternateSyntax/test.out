1
2.34
0
HAI
1
2.34
0
HAI
9
8.76
1
BAI
1
2.34
0
HAI
9
8.76
1
BAI
1
2.34
0
HAI
